src/lib/layout/layout.ts

```diff
@@ -105,7 +105,17 @@ export async function create(optionsGiven?: Options): Promise<Layout> {
     cwd: optionsGiven?.cwd ?? process.cwd(),
   }
   const data = await scan({ cwd: options.cwd })
-  return createFromData({ ...data, buildOutput: options.buildOutput })
+  const layout = createFromData({ ...data, buildOutput: options.buildOutput })
+
+  /**
+   * Save the created layout in the env
+   */
+  process.env = {
+    ...process.env,
+    ...saveDataForChildProcess(layout),
+  }
```

Comment:
Not sure if this is safe. We still need to pass the `NEXUS_LAYOUT` in `generateArtifacts` in the build command. So instead of passing it manually, I decided to make `Layout.create` always save "itself" to the environment. Do you see any reason not to do that?

Reply:
Let's try